quote-generator/src/app/app.functions/create-quote.js

```diff
@@ -0,0 +1,98 @@
+// for HubSpot API calls
+const hubspot = require('@hubspot/api-client');
+
+const hubspotClient = new hubspot.Client({
+  accessToken: process.env['PRIVATE_APP_ACCESS_TOKEN'],
+});
+
+exports.main = async (context = {}, sendResponse) => {
+  const { hs_object_id } = context.propertiesToSend;
+  const { distance, sku, numberOfBuses, quoteName } = context.event.payload;
+
+  const product = await findProductBySKU(sku);
+
+  if (product === null) {
+    sendResponse({ error: 'PRODUCT_NOT_FOUND' });
+  } else {
+    const quote = await createQuote({
+      dealId: hs_object_id,
+      quoteName,
+    });
+    for (let i = 0; i < numberOfBuses; i++) {
+      await addLineItem({
+        productId: product.id,
+        quoteId: quote.id,
+        quantity: distance,
+      });
+    }
+    sendResponse({ quote });
+  }
+};
+
+async function createQuote({ dealId, quoteName }) {
+  const oneWeek = 168 * 60 * 60000;
```

Comment:
nit: consider placing each of those numbers in a constant that has a descriptive name. For example `const MIN_IN_HOUR=60`